tests/framework/console/ControllerTest.php

```diff
@@ -89,4 +94,54 @@ public function testResponse()
         $this->assertResponseStatus($status, $response);
     }
 
+    /**
+     * @see https://github.com/yiisoft/yii2/issues/12028
+     */
+    public function testHelpOption()
+    {
+        Yii::$app->controllerMap = [
+            'fake' => 'yiiunit\framework\console\FakeController',
+            'help' => 'yiiunit\framework\console\FakeHelpController',
+        ];
+
+        $controller = new FakeController('posts', Yii::$app);
+        $controller->runAction('index');
+
+        $this->assertTrue(FakeController::getWasActionIndexCalled());
+        $this->assertNull(FakeHelpController::getActionIndexLastCallParams());
+
+        $controller = new FakeController('posts', Yii::$app);
```

Comment:
'crucial'....it depends on which hat you put on. From the consistency & readability & stability perspective, it is a better practice.